ReactiveObjC/RACNeverSignal.m

```diff
@@ -0,0 +1,63 @@
+//
+//  RACNeverSignal.m
+//  ReactiveObjC-macOS
+//
+//  Created by Yaron Inger on 19/12/2018.
+//  Copyright (c) 2018 GitHub, Inc. All rights reserved.
+//
+
+#import "RACNeverSignal.h"
+
+#import <ReactiveObjC/EXTScope.h>
+
+#import "RACCompoundDisposable.h"
+#import "RACSubscriber.h"
+
+@interface RACNeverSignal ()
+
+/// Array that holds the subscribers to avoid them disposing the subscription.
+@property (nonatomic, readonly) NSMutableArray<id<RACSubscriber>> *subscribers;
+
+@end
+
+@implementation RACNeverSignal
+
+- (instancetype)init {
+  if (self = [super init]) {
+    _subscribers = [NSMutableArray array];
+  }
+  return self;
+}
+
++ (RACSignal *)never {
+  return [[[self alloc] init] setNameWithFormat:@"+never"];
+}
+
+- (nullable RACDisposable *)subscribe:(id<RACSubscriber>)subscriber {
+  NSCParameterAssert(subscriber != nil);
+
+  NSMutableArray *subscribers = self.subscribers;
+  @synchronized (subscribers) {
+    [subscribers addObject:subscriber];
+  }
+
+  RACCompoundDisposable *disposable = [RACCompoundDisposable compoundDisposable];
+
+  @rac_weakify(subscribers, subscriber);
+  [disposable addDisposable:[RACDisposable disposableWithBlock:^{
+    @rac_strongify(subscribers, subscriber);
```

Comment:
what prevents now from `subscribers` to be deallocated when the signal is deallocated?

Reply:
In other words: do we want the subscription lifetime to be bound to the signal lifetime? This is not the case with dynamic signals. But if `subscribers` was retained in the desposable we will get a temporary retain cycle:
subscribers -> subscriber -> disposable -> subscribers.
Then the cycle would break when removing the `subscriber` from the array and then `subscribers` would get deallocated only when both the signal was deallocated and all running subscriptions were disposed (which in a way makes more sense to me).

Reply:
@ofirgluzman I agree with you, and added another commit + changed the test such that if `RACNeverSignal` will dealloc, the disposable will not get disposed, but only on manual disposal / `-takeUntil:`.

Would be happy to get more eyes on this as this is the thorny issue about this PR.